crates/bevy_reflect/src/fields.rs

```diff
@@ -0,0 +1,79 @@
+use crate::Reflect;
+use std::any::TypeId;
+use std::borrow::{Borrow, Cow};
+
+/// The named field of a reflected struct
+#[derive(Clone, Debug)]
+pub struct NamedField {
+    name: Cow<'static, str>,
+    type_name: Cow<'static, str>,
+    type_id: TypeId,
+}
+
+impl NamedField {
+    pub fn new<T: Reflect>(name: &str) -> Self {
+        Self {
+            name: Cow::Owned(name.into()),
+            type_name: Cow::Owned(std::any::type_name::<T>().to_string()),
+            type_id: TypeId::of::<T>(),
+        }
+    }
+
+    /// The name of the field
+    pub fn name(&self) -> &str {
+        self.name.borrow()
+    }
+
+    /// The type name of the field
+    pub fn type_name(&self) -> &str {
+        self.type_name.borrow()
+    }
+
+    /// The `TypeId` of the field
```

Comment:
```suggestion
    /// The [`TypeId`] of the field
```

Reply:
Does this work on external types? Or is it because it's part of the std lib that it works?